lightning/src/ln/interactivetxs.rs

```diff
@@ -1062,6 +1320,39 @@ pub(crate) enum InteractiveTxMessageSend {
 	TxComplete(msgs::TxComplete),
 }
 
+impl InteractiveTxMessageSend {
+	pub fn into_msg_send_event(self, counterparty_node_id: PublicKey) -> MessageSendEvent {
+		match self {
+			InteractiveTxMessageSend::TxAddInput(msg) => {
+				MessageSendEvent::SendTxAddInput { node_id: counterparty_node_id, msg }
+			},
+			InteractiveTxMessageSend::TxAddOutput(msg) => {
+				MessageSendEvent::SendTxAddOutput { node_id: counterparty_node_id, msg }
+			},
+			InteractiveTxMessageSend::TxComplete(msg) => {
+				MessageSendEvent::SendTxComplete { node_id: counterparty_node_id, msg }
+			},
+		}
+	}
+}
+
+pub(super) struct InteractiveTxMessageSendResult(
+	pub Result<InteractiveTxMessageSend, msgs::TxAbort>,
+);
+
+impl InteractiveTxMessageSendResult {
+	pub fn into_msg_send_event(self, counterparty_node_id: PublicKey) -> MessageSendEvent {
+		match self.0 {
+			Ok(interactive_tx_msg_send) => {
+				interactive_tx_msg_send.into_msg_send_event(counterparty_node_id)
+			},
+			Err(tx_abort_msg) => {
+				MessageSendEvent::SendTxAbort { node_id: counterparty_node_id, msg: tx_abort_msg }
+			},
+		}
+	}
+}
```

Comment:
Shouldn't hold up this PR, but is there any reason why we don't simply add a `TxAbort` variant to `InteractiveTxMessageSend`? Then we wouldn't need either `InteractiveTxMessageSendResult` or `HandleTxCompleteResult`, IIUC, which would remove two layers of structs.

Otherwise, we go through the following conversions just to end up with a `MessageSendEvent`:
- `Result<InteractiveTxMessageSend, AbortReason>`
- `InteractiveTxMessageSendResult(Result<InteractiveTxMessageSend, msgs::TxAbort>)`
- `MessageSendEvent`

The `Result`s don't seem to buy us anything more than the extra variant would as we aren't doing anything special for the `Err` case. Then it would simply be:
- `InteractiveTxMessageSend`
- `MessageSendEvent`

And AFAICT, `InteractiveTxConstructor` has the `channel_id`, so there's no need to grab it from the `ChannelContext`. Or will it change in some case?

Reply:
Yes, this would work. I imagine the original iteration I had in mind has some `Err` case to handle which is not applicable anymore.

Yeah the channel_id is fixed once we have a constructor. I'm quite sure the same is true of splicing, but it can be refactored if that's not the case.

Reply:
I'm going to leave this as is for now and just apply your other feedback. `TxAbort` is currently handled distinctly from `InteractiveTxMessageSend`. It is handled at the `Channel` level. I suspect we can have it handled that same as any other variant of `InteratveTxMessageSend` without issues but it would require some refactoring. Will create a tracking issue and leave for now.